islands/ItemsList.tsx

```diff
@@ -0,0 +1,90 @@
+// Copyright 2023 the Deno authors. All rights reserved. MIT license.
+import { useComputed, useSignal } from "@preact/signals";
+import { useEffect } from "preact/hooks";
+import type { Item } from "@/utils/db.ts";
+import { LINK_STYLES } from "@/utils/constants.ts";
+import IconInfo from "tabler_icons_tsx/info-circle.tsx";
+import ItemSummary from "@/components/ItemSummary.tsx";
+import { fetchValues } from "@/utils/http.ts";
+
+async function fetchVotedItems() {
+  const url = "/api/me/votes";
+  const resp = await fetch(url);
+  if (!resp.ok) throw new Error(`Request failed: GET ${url}`);
+  return await resp.json() as Item[];
+}
+
+function EmptyItemsList() {
+  return (
+    <>
+      <div class="flex flex-col justify-center items-center gap-2">
+        <div class="flex flex-col items-center gap-2 pt-16">
+          <IconInfo class="w-10 h-10 text-gray-400 dark:text-gray-600" />
+          <p class="text-center font-medium">No items found</p>
+        </div>
+
+        <a
+          href="/submit"
+          class="inline-flex items-center justify-center gap-2 rounded-md px-3 py-2 text-primary hover:underline"
+        >
+          Submit your project
+        </a>
+      </div>
+    </>
+  );
+}
+
+export default function ItemsList(props: { endpoint: string }) {
+  const itemsSig = useSignal<Item[]>([]);
+  const votedItemsIdsSig = useSignal<string[]>([]);
+  const cursorSig = useSignal("");
+  const isLoadingSig = useSignal(false);
```

Comment:
I'm curious why `useSignal` is used here instead of `useState`

Reply:
Good question! For context: https://preactjs.com/blog/introducing-signals/